### PR TITLE
DNMY: [HiGHS] build on experimental platforms

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

File: H/HiGHS/build_tarballs.jl [hunk 1]
```diff
@@ -4,7 +4,7 @@ using BinaryBuilder, Pkg
 
 name = "HiGHS"
 
-version = v"1.1.0"
+version = v"1.1.1"
```
Comment (giordano) [inline H/HiGHS/build_tarballs.jl:7]: The source doesn't change?

Comment (odow) [reply H/HiGHS/build_tarballs.jl:7]: I was just going to rebuild what we had. But I guess it doesn't hurt to bump to the latest `master`. They still aren't doing versioned releases upstream.

**Comment by odow** [comment]
@lgottwald looks like there are some issues compiling your recent parallel changes https://github.com/ERGO-Code/HiGHS/pull/611? The diff is quite big so I didn't look through it in detail. Do we need to modify our build script in some way?

cc @jajhall

**Comment by lgottwald** [comment]
It seems some type aliases are not provided by older gcc, I'll add a workaround for that then we'll see what errors are left

File: H/HiGHS/build_tarballs.jl [hunk 2]
```diff
@@ -4,12 +4,12 @@ using BinaryBuilder, Pkg
...
 
 sources = [
     GitSource(
         "https://github.com/ERGO-Code/HiGHS.git",
-        "bd9703cc1a896c68473144064011cc9d64246e07",
+        "6c2ab5414fd9670ad961920aefe0c57d082eb6cc",
```
Comment (odow) [inline H/HiGHS/build_tarballs.jl:12]: @lgottwald I bumped the commit to your PR. The other option we have is to use a newer version of GCC, although it's preferred to build on the oldest possible to maximize compatibility.

**Comment by lgottwald** [comment]
Some errors are gone and I now looked at some others. I now (hopefully) fixed some remaining errors and warnings. Once the build in the HiGHS PR passes you could maybe trigger another Julia build.

**Comment by lgottwald** [comment]
added another workaround for old gcc compilers to master

**Comment by odow** [comment]
Let's see if that worked, otherwise I will bump the minimum version of GCC.

**Comment by lgottwald** [comment]
Interesting. Fewer errors but the ones that are left are getting weirder. There are now internal compiler errors and linker errors that I cannot really address by changing the code.

**Comment by lgottwald** [comment]
It seems like AArch64 build fail due to a compiler bug and mingw builds from a linker error. The linker error could probably be fixed if things worked before and is not really related to the parallel code. The AArch64 failure seems to be some kind of gcc code generation bug for atomics and likely requires to update the compiler.

File: H/HiGHS/build_tarballs.jl [hunk 3]
```diff
@@ -72,5 +72,6 @@ build_tarballs(
     platforms,
     products,
     dependencies;
-    preferred_gcc_version = v"4.9",
+    preferred_gcc_version = v"5",
```
Comment (giordano) [inline H/HiGHS/build_tarballs.jl:75]: You may want to go to 6
```suggestion
    preferred_gcc_version = v"6",
```
We have 4.8, 5, etc..., so between 4.9 and 5 it doesn't change much :wink:

**Comment by odow** [comment]
@lgottwald, now just the windows error, which I assume is some missing DLLEXPORT stuff?

You can probably revert your GCC <5 commit: https://github.com/ERGO-Code/HiGHS/commit/16f971a4c95dcfd06796b63ebb2f0961d357d8e2

**Comment by lgottwald** [comment]
I need to look further into the linker error tomorrow. If support for gcc 4.8 is not required anymore I can also remove some other workarounds but I'll leave them for the moment

**Pull request closed without merging**